types/views/views.go

```diff
@@ -88,6 +88,36 @@ func (v SliceView[T, V]) AsSlice() []V {
 	return v.AppendTo(nil)
 }
 
+func (v SliceView[T, V]) Iterator() SliceViewIterator[T, V] {
+	return SliceViewIterator[T, V]{}
+}
+
+// SliceViewIterator provides an iterator interface over a SliceView.
+//
+// A typical usage example is
+//   var s Slice[T]
+//   for i := s.Iterator(); i.Valid(); i = i.Next() {
+//     v := i.Val()
+//   }
+type SliceViewIterator[T ViewCloner[T, V], V StructView[T]] struct {
```

Comment:
It is a bit weird that the iterator patterns for slice here and in `Map.Range` are drastically different. I don't know how to unify the two in a way that performs well since you can't easily capture the iteration state of a map in a variable.